package/kedro_viz/server.py

```diff
@@ -195,46 +195,102 @@ def pretty_name(name):
 
     nodes = []
     edges = []
-    namespace_tags = defaultdict(set)
-    all_tags = set()
-
-    for node in sorted(pipeline.nodes, key=lambda n: n.name):
-        task_id = _hash(str(node))
-        nodes.append(
-            {
-                "type": "task",
-                "id": task_id,
-                "name": getattr(node, "short_name", node.name),
-                "full_name": getattr(node, "_func_name", str(node)),
-                "tags": sorted(node.tags),
-            }
-        )
-        all_tags.update(node.tags)
+
+    from toposort import toposort, toposort_flatten
+    from collections import namedtuple
+
+    catalog_metadata = catalog._data_sets
+
+    layer_dependencies = defaultdict(set)
+    datasets_by_layers = defaultdict(set)
+    dependencies = defaultdict(set)
+
+    for node in pipeline.nodes:
+        dependencies[node._func_name].update(node.inputs)
+
+        input_layers = set()
         for data_set in node.inputs:
-            namespace = data_set.split("@")[0]
-            edges.append({"source": _hash(namespace), "target": task_id})
-            namespace_tags[namespace].update(node.tags)
-        for data_set in node.outputs:
-            namespace = data_set.split("@")[0]
-            edges.append({"source": task_id, "target": _hash(namespace)})
-            namespace_tags[namespace].update(node.tags)
-
-    for namespace, tags in sorted(namespace_tags.items()):
-        is_param = bool("param" in namespace.lower())
-        nodes.append(
-            {
-                "type": "parameters" if is_param else "data",
-                "id": _hash(namespace),
-                "name": pretty_name(namespace),
-                "full_name": namespace,
-                "tags": sorted(tags),
-            }
-        )
+            if data_set.startswith("param"):
```

Comment:
What layer should be used for params?

Reply:
If the dataset does not have a layer it should still be ranked - thee logic stays the same. The same applies to params, they are treated as datasets without layer.